server/db/mysql/adapter.go

```diff
@@ -65,16 +74,21 @@ func (a *adapter) Open(jsonconfig json.RawMessage) error {
 		return errors.New("mysql adapter failed to parse config: " + err.Error())
 	}
 
-	a.dsn = config.DSN
-	if a.dsn == "" {
-		a.dsn = defaultDSN
+	if config.DBName == "" {
+		config.DBName = defaultDatabase
 	}
-
 	a.dbName = config.DBName
-	if a.dbName == "" {
-		a.dbName = defaultDatabase
+
+	if config.DSN != "" {
+		return errors.New(fmt.Sprintf(mysqlConfigErrorText, "dsn"))
```

Comment:
`errors.New(fmt.Sprintf(...))` -> `fmt.Errorf(...)`

I don't see `%s` in `mysqlConfigErrorText`, so `errors.New(mysqlConfigErrorText)` would produce the same result.